include/mbedtls/ecp.h

```diff
@@ -53,139 +64,145 @@ extern "C" {
  *                  The point is zero, or "at infinity", if Z == 0.
  *                  Otherwise, X and Y are its standard (affine) coordinates.
  */
 typedef struct
 {
-    mbedtls_mpi X;          /*!<  the point's X coordinate  */
-    mbedtls_mpi Y;          /*!<  the point's Y coordinate  */
-    mbedtls_mpi Z;          /*!<  the point's Z coordinate  */
+    mbedtls_mpi X;          /*!< The X coordinate of the ECP point. */
+    mbedtls_mpi Y;          /*!< The Y coordinate of the ECP point. */
+    mbedtls_mpi Z;          /*!< The Z coordinate of the ECP point. */
 }
 mbedtls_ecp_point;
 
 /**
- * \brief           ECP group structure
+ * \brief           The ECP group structure.
+ *
+ * We consider two types of curve equations:
+ * <ul><li>Short Weierstrass: y^2 = x^3 + \p A x + \p B     mod P   
+ * (SEC1 + RFC-4492)</li>
+ * <li>Montgomery: y^2 = x^3 + A x^2 + x mod P (Curve25519 + draft)</li></ul>
+ * In both cases, the generator (G) for a prime-order subgroup is fixed.
  *
- * We consider two types of curves equations:
- * 1. Short Weierstrass y^2 = x^3 + A x + B     mod P   (SEC1 + RFC 4492)
- * 2. Montgomery,       y^2 = x^3 + A x^2 + x   mod P   (Curve25519 + draft)
- * In both cases, a generator G for a prime-order subgroup is fixed. In the
- * short weierstrass, this subgroup is actually the whole curve, and its
- * cardinal is denoted by N.
+ * For Short Weierstrass, this subgroup is the whole curve, and its
+ * cardinal is denoted by \p N. Our code requires that \p N is an odd prime.
  *
- * In the case of Short Weierstrass curves, our code requires that N is an odd
- * prime. (Use odd in mbedtls_ecp_mul() and prime in mbedtls_ecdsa_sign() for blinding.)
+ * \note For blinding, use odd in mbedtls_ecp_mul() and prime in
+ * mbedtls_ecdsa_sign().
  *
- * In the case of Montgomery curves, we don't store A but (A + 2) / 4 which is
- * the quantity actually used in the formulas. Also, nbits is not the size of N
- * but the required size for private keys.
+ * For Montgomery curves, we do not store \p A, but (A + 2) / 4, which is 
+ * the quantity used in the formulas. Additionally, \p nbits is not the 
+ * size of \p N but the required size for private keys.
  *
- * If modp is NULL, reduction modulo P is done using a generic algorithm.
- * Otherwise, it must point to a function that takes an mbedtls_mpi in the range
- * 0..2^(2*pbits)-1 and transforms it in-place in an integer of little more
- * than pbits, so that the integer may be efficiently brought in the 0..P-1
- * range by a few additions or substractions. It must return 0 on success and
- * non-zero on failure.
+ * If \p modp is NULL, reduction modulo \p P is done using a generic algorithm. 
+ * Otherwise, it must point to a function that takes an \p mbedtls_mpi in the 
+ * range of 0..2^(2*pbits)-1, and transforms it in-place in an integer of 
+ * little more than \p pbits, so that the integer may be efficiently brought 
```

Comment:
Pre-existing: `... and transforms it in-place to an integer which is congruent mod \p P to the given MPI and is close enough to \p pbits in size, so that it may be efficiently brought in the 0..P-1 range by a few additions or subtractions (hence it's only an approximative modular reduction).`